plugins/core/IO/qLASIO/include/LasDetails.h

```diff
@@ -0,0 +1,356 @@
+constexpr const char *ScanDirectionFlag = "Scan Direction Flag";
+constexpr const char *EdgeOfFlightLine = "EdgeOfFlightLine";
+constexpr const char *Classification = "Classification";
+constexpr const char *SyntheticFlag = "Synthetic Flag";
+constexpr const char *KeypointFlag = "Keypoint Flag";
+constexpr const char *WithheldFlag = "Withheld Flag";
+constexpr const char *ScanAngleRank = "Scan Angle Rank";
+constexpr const char *UserData = "User Data";
+constexpr const char *PointSourceId = "Point Source ID";
+constexpr const char *GpsTime = "Gps Time";
+
+// 1.4 point format 6 stuff
+constexpr const char *ScanAngle = "Scan Angle";
+constexpr const char *ScannerChannel = "Scanner Channel";
+constexpr const char *OverlapFlag = "Overlap Flag";
+constexpr const char *NearInfrared = "Near Infrared";
+} // namespace LasNames
+
+/// class used to link a LAS field defined by the LAS standard to the CloudCompare
+/// scalar field that stores or will store the values.
+struct LasScalarField
+{
+    /// Enum used to uniquely identify LAS fields
+    /// with a clear distinction between 'normal' and 'extended' fields
+    /// which may have the same name (e.g. Classification)
+    enum Id
+    {
+        Intensity = 0,
+        ReturnNumber = 1,
+        NumberOfReturns = 2,
+        ScanDirectionFlag,
+        EdgeOfFlightLine,
+        Classification,
+        SyntheticFlag,
+        KeypointFlag,
+        WithheldFlag,
+        ScanAngleRank,
+        UserData,
+        PointSourceId,
+        GpsTime,
+        // Extended (LAS 1.4)
+        ExtendedScanAngle,
+        ExtendedScannerChannel,
+        OverlapFlag,
+        ExtendedClassification,
+        ExtendedReturnNumber,
+        ExtendedNumberOfReturns,
+        NearInfrared
+    };
+
+    struct Range
+    {
+
+        template <class T>
+        constexpr Range(T min_, T max_)
+            : min(static_cast<ScalarType>(min_)), max(static_cast<ScalarType>(max_))
+        {
+        }
+
+        template <class T> static constexpr Range ForType()
+        {
+            return Range(std::numeric_limits<T>::min(), std::numeric_limits<T>::max());
+        }
+
+        static Range ForBitSize(uint8_t numBits)
+        {
+            Range range(0.0, 0.0);
+            range.max = static_cast<ScalarType>(std::pow(2, numBits) - 1.0);
+            return range;
+        }
+
+        ScalarType min;
+        ScalarType max;
+    };
+
+  public: // Methods and Constructors
+    explicit LasScalarField(LasScalarField::Id id, ccScalarField *sf = nullptr);
+
+    const char *name() const;
+
+  public: // Static functions
+    static constexpr const char *NameFromId(LasScalarField::Id id);
+    static LasScalarField::Id IdFromName(const char *name, unsigned int targetPointFormat);
+    static LasScalarField::Range ValueRange(LasScalarField::Id id);
+
+    // TODO They should be private
+  public: // Members
+    /// The Id of the LAS field this relates to.
+    Id id;
```

Comment:
default initialize to something? Or to an 'invalid' enum?

Reply:
I explicitely disabled the default constructor of the `LasScalarField` class, which means that to create an instance of it
one must use the constructor that initializes this id to a proper value.

Otherwise if I were to introduce an "Invalid" variant I'd have to check at runtime for this variant in a bunch of places,
so it seems better to me to not allow an invalid / un-initialized value by deleting the default ctor

Reply:
Ah ok, it was just surprising to see that it was the only member that was not default initialized.